In [ ]:
test_data = ["One of the other reviewers has mentioned that after watching just 1 Oz episode you'll be hooked. They are right, as this is exactly what happened with me. Would definitely recommend this movie",
             "A wonderful little production. The filming technique is very unassuming- very old-time-BBC fashion and gives a comforting, and sometimes discomforting, sense of realism to the entire piece.",
             "I thought this was a wonderful way to spend time on a too hot summer weekend, sitting in the air conditioned theater and watching a light-hearted comedy. The plot is simplistic, but the dialogue is witty and the characters are likable.",
             "Basically there's a family where a little boy (Jake) thinks there's a zombie in his closet & his parents are fighting all the time. This movie is slower than a soap opera... and suddenly, Jake decides to become Rambo and kill the zombie. A film to forget.",
             "This is a movie that seems to be telling us what money, power and success do to people in the different situations we encounter. Stylishly, the film has a sophisticated",
             "Probably my all-time favorite movie, a story of selflessness, sacrifice and dedication to a noble cause, but it's not preachy or boring. It just never gets old, despite my having seen it some 15 or more times in the last 25 years.",
             "I sure would like to see a resurrection of a up dated Seahunt series with the tech they have today it would bring back the kid excitement in me.I grew up on black and white TV and Seahunt with Gunsmoke were my hero's every week.",
             "This show was an amazing, fresh & innovative idea in the 70's when it first aired. The first 7 or 8 years were brilliant, but things dropped off after that. By 1990, the show was not really funny anymore, and it's continued its decline further to the complete waste of time it is today.",
             "Encouraged by the positive comments about this film on here I was looking forward to watching this film. Bad mistake. I've seen 950+ films and this is truly one of the worst of them - it's awful in almost every way: editing, pacing, storyline, 'acting,' and soundtrack.",
             "If you like original gut wrenching laughter you will like this movie. If you are young or old then you will love this movie, hell even my mom liked it. Great Camp!!!",
             "Phil the Alien is one of those quirky films where the humour is based around the oddness of everything rather than actual punchlines. At first it was very odd and pretty funny but as the movie progressed I didn't find the jokes or oddness funny anymore.",
             "I saw Psycho when it came out and I loved that the star, Janet Leigh, was bumped off early in the film. I sat up and took notice at that point. Since screenwriters are making up the story, make it up to be as scary as possible and not from a well-worn formula. There are no rules.",
             "If you wanna stay true to Bolls shenanigans then go and see this movie you will not be disappointed it delivers the true Boll experience, meaning most of it will suck.",
             "The cast played Shakespeare. Shakespeare lost. I appreciate that this is trying to bring Shakespeare to the masses, but why ruin something so good.",
             "This a fantastic movie of three prisoners who become famous. One of the actors is george clooney and I'm not a fan but this roll is not bad. Another good thing about the movie is the soundtrack. I recommand this movie to everybody. Greetings Bart",
             "Kind of drawn in by the erotic scenes, only to realize this was one of the most amateurish and unbelievable bits of film I've ever seen. Sort of like a high school film project.",
             "Some films just simply should not be remade. This is one of them. In and of itself it is not a bad film. But it fails to capture the flavor and the terror of the 1963 film of the same title.",
             "This movie made it into one of my top 10 most awful movies. Horrible. There wasn't a continuous minute where there wasn't a fight with one monster or another.",
             "I remember this film,it was the first film i had watched at the cinema the picture was dark in places i was very nervous it was back in 74/75 my Dad took me my brother & sister to Newbury cinema in Newbury Berkshire England. If anyone knows where to find this on DVD etc please let me know.",
             "An awful film! It must have been up against some real stinkers to be nominated for the Golden Globe. They've taken the story of the first famous female Renaissance painter and mangled it beyond recognition."]

In [ ]:
test_data_labels = [1,1,1,0,1,1,1,0,0,1,0,0,0,0,1,0,1,0,1,0]

In [ ]:
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.tree import DecisionTreeClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

vectorizer = TfidfVectorizer()
X = vectorizer.fit_transform(test_data)

X_train, X_test, y_train, y_test = train_test_split(X, test_data_labels, test_size=0.75, random_state=42)

dt_classifier = DecisionTreeClassifier()
dt_classifier.fit(X_train, y_train)

lr_classifier = LogisticRegression()
lr_classifier.fit(X_train, y_train)

y_pred_dt = dt_classifier.predict(X_test)
y_pred_lr = lr_classifier.predict(X_test)

accuracy_dt = accuracy_score(y_test, y_pred_dt)
accuracy_lr = accuracy_score(y_test, y_pred_lr)

print("Decision Tree Classifier Accuracy:", accuracy_dt)
print("Logistic Regression Accuracy:", accuracy_lr)
print(y_pred_dt)

Decision Tree Classifier Accuracy: 0.5333333333333333
Logistic Regression Accuracy: 0.4666666666666667
[1 0 1 1 0 0 0 1 0 0 1 1 0 0 1]


The two algorithms I have chose are logistic regression and decision trees. The two algorithms were both fitted to the dataset and its values. Their accuracy scores are: Logistic Regression: 0.467 Decision Tree: 0.533

In [ ]:
from sklearn.metrics import confusion_matrix

correct_both = []
incorrect_both = []
correct_dt_incorrect_lr = []
correct_lr_incorrect_dt = []

for i in range(len(y_pred_dt)):
    label = y_test[i]
    prediction_dt = y_pred_dt[i]
    prediction_lr = y_pred_lr[i]
    text = test_data[i]

    if label == prediction_dt and label == prediction_lr:
        correct_both.append((text, label))
    elif label != prediction_dt and label != prediction_lr:
        incorrect_both.append((text, label, prediction_dt, prediction_lr))
    elif label == prediction_dt and label != prediction_lr:
        correct_dt_incorrect_lr.append((text, label, prediction_lr))
    elif label == prediction_lr and label != prediction_dt:
        correct_lr_incorrect_dt.append((text, label, prediction_dt))

print("Decision Tree Classifier Accuracy:", accuracy_dt)
print("Confusion Matrix for Decision Tree Classifier:")
print(confusion_matrix(y_test, y_pred_dt))
print("\nLogistic Regression Accuracy:", accuracy_lr)
print("Confusion Matrix for Logistic Regression:")
print(confusion_matrix(y_test, y_pred_lr))

print("\nCases that both models classified correctly:")
for text, label in correct_both:
    print(f"Text: {text}, Actual Label: {label}")

print("\nCases that both models classified incorrectly:")
for text, actual, dt_prediction, lr_prediction in incorrect_both:
    print(f"Text: {text}, Actual Label: {actual}, DT Prediction: {dt_prediction}, LR Prediction: {lr_prediction}")

print("\nCases that were correctly classified by Decision Tree and incorrectly by Logistic Regression:")
for text, actual, lr_prediction in correct_dt_incorrect_lr:
    print(f"Text: {text}, Actual Label: {actual}, LR Prediction: {lr_prediction}")

print("\nCases that were correctly classified by Logistic Regression and incorrectly by Decision Tree:")
for text, actual, dt_prediction in correct_lr_incorrect_dt:
    print(f"Text: {text}, Actual Label: {actual}, DT Prediction: {dt_prediction}")



Decision Tree Classifier Accuracy: 0.5333333333333333
Confusion Matrix for Decision Tree Classifier:
[[4 3]
 [4 4]]

Logistic Regression Accuracy: 0.4666666666666667
Confusion Matrix for Logistic Regression:
[[7 0]
 [8 0]]

Cases that both models classified correctly:
Text: A wonderful little production. The filming technique is very unassuming- very old-time-BBC fashion and gives a comforting, and sometimes discomforting, sense of realism to the entire piece., Actual Label: 0
Text: This is a movie that seems to be telling us what money, power and success do to people in the different situations we encounter. Stylishly, the film has a sophisticated, Actual Label: 0
Text: I sure would like to see a resurrection of a up dated Seahunt series with the tech they have today it would bring back the kid excitement in me.I grew up on black and white TV and Seahunt with Gunsmoke were my hero's every week., Actual Label: 0
Text: The cast played Shakespeare. Shakespeare lost. I appreciate that thi

Both models are very similar as we can see with the cases above which also reflects in the similar accuracy score.